tiledb/api/src/query/read/callback.rs

```diff
@@ -209,6 +209,35 @@ mod impls {
         }
     }
 
+    impl<C> ReadCallback for (Vec<C>, Vec<u8>)
+    where
+        C: CAPISameRepr,
+    {
+        type Unit = C;
+        type Intermediate = ();
+        type Final = Self;
+        type Error = std::convert::Infallible;
+
+        fn intermediate_result(
+            &mut self,
+            arg: RawReadOutput<Self::Unit>,
+        ) -> Result<Self::Intermediate, Self::Error> {
+            self.0
+                .extend_from_slice(&arg.input.data.as_ref()[0..arg.nvalues]);
+            self.1.extend_from_slice(
+                &arg.input.validity.as_ref().unwrap().as_ref()[0..arg.nvalues],
```

Comment:
`unwrap` here should have a comment.  If the user supplied the wrong kind of buffer, would it be caught by now?  The answer is "yes", right, because the query won't run if you have the wrong `Option`?